api/contrib/envoy/extensions/filters/http/checksum/v3alpha/checksum.proto

```diff
@@ -0,0 +1,36 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.checksum.v3alpha;
+
+import "envoy/type/matcher/v3/string.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.checksum.v3alpha";
+option java_outer_classname = "ChecksumProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/checksum/v3alpha";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Checksum HTTP filter]
+//
+// Filter to reject responses that don't match a specified checksum.
+// To avoid holding the entire response in memory, the rejection occurs at the end of the stream.
+// [#extension: envoy.filters.http.checksum]
+
+message ChecksumConfig {
+  message Checksum {
+    type.matcher.v3.StringMatcher path_matcher = 1;
+
+    string sha256 = 2;
```

Comment:
Please add the PGV validations and more detailed comments, esp for the `path_patcher`.

And I wonder why `StringMatcher` is used here? Because only one sha256 is used here, so seem extact much should be enough for it?

Reply:
i think i just prototyped it that way - there was some reasoning behind using matching - but i guess not for the path - so im good with exact match

Reply:
One suggested reason was if you might be trying multiple mirrors, you might have e.g.
`https://github.com/path/to/file.zip`
`https://mirrors.foobar.com/github.com/path/to/file.zip`
in a retry sequence, in which case a matcher accepting `^.*/path/to/file.zip$` would be more convenient than specifying the same checksum repeatedly.

At this point I prefer the `StringMatcher` because it's already implemented and tested that way, it doesn't add much complexity, and the flexibility *might* have a use.

Reply:
sgtm.

The unique drawback is we cannot find the checksum by a simple hash searching and need to iterate all matchers. 

So, my final suggesion is to move the path_matcher to an `oneof`. Then, in the future oneday if there are lots of matchers and we want to do some optimizations, we can add some other matcher.

Reply:
Yeah, agree on that drawback (we *could* make it a hash lookup of exact matches and then fall back to O(n) for other matchers, with the same API, but for the intended use case right now it's not worth the effort to optimize. The annoyance there is with the documented "resolved in specified order", changing to "exact matches resolved first then the rest are resolved in specified order" would still be technically an API change!)